library/Requests/Transport/cURL.php

```diff
@@ -296,15 +311,18 @@ protected function setup_handle($url, $headers, $data, $options) {
 				break;
 			case Requests::PATCH:
 			case Requests::PUT:
+			case Requests::DELETE:
+			case Requests::OPTIONS:
 				curl_setopt($this->handle, CURLOPT_CUSTOMREQUEST, $options['type']);
 				curl_setopt($this->handle, CURLOPT_POSTFIELDS, $data);
 				break;
-			case Requests::DELETE:
-				curl_setopt($this->handle, CURLOPT_CUSTOMREQUEST, 'DELETE');
-				break;
 			case Requests::HEAD:
+				curl_setopt($this->handle, CURLOPT_CUSTOMREQUEST, $options['type']);
 				curl_setopt($this->handle, CURLOPT_NOBODY, true);
```

Comment:
@egeloen Looks like `CURLOPT_POSTFIELDS` doesn't get set for HEAD requests; was this intentional?